src/lib/y2users/collision_checker.rb

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2issues"
+
+module Y2Users
+  # Internal class to check if there are collisions in config.
+  # This is not part of the stable API.
+  class CollisionChecker
+    include Yast::I18n
+
+    # Issue location describing the Group#gid attribute
+    GROUP_ID_LOC = "group:gid".freeze
+    # Issue location describing the Group#name attribute
+    GROUP_NAME_LOC = "group:name".freeze
+    # Issue location describing the User#uid attribute
+    USER_ID_LOC = "user:uid".freeze
+    # Issue location describing the User#uid attribute
+    USER_NAME_LOC = "user:name".freeze
+    private_constant :GROUP_ID_LOC, :GROUP_NAME_LOC, :USER_ID_LOC, :USER_NAME_LOC
+
+    # Constructor
+    #
+    # @param config [Y2Users::Config] config to detect colliding ids
+    def initialize(config)
+      textdomain "users"
+      @config = config
+    end
+
+    # Returns a list of issues found while checking collisions
+    #
+    # @return [Y2Issues::List]
+    def issues
+      list = Y2Issues::List.new
+
+      list.concat(duplicite_users_ids)
+      list.concat(duplicite_users_names)
+      list.concat(duplicite_groups_ids)
+      list.concat(duplicite_groups_names)
+
+      list
+    end
+
+  private
+
+    # @return [Y2Users::Config] config to validate
+    attr_reader :config
+
+    def duplicite_users_ids
```

Comment:
duplicite? Is that even an English word?

Reply:
french one, I will fix it.